tests/test_responses.py

```diff
@@ -366,3 +367,20 @@ def test_streaming_response_known_size(test_client_factory):
     client: TestClient = test_client_factory(app)
     response = client.get("/")
     assert response.headers["content-length"] == "10"
+
+
+@pytest.mark.parametrize("status_code", (100, 204, 304))
+def test_no_content_length(
+    status_code: int, test_client_factory: Callable[..., TestClient]
+):
+    app = Response(content=b"non_empty", status_code=status_code)
+    client = test_client_factory(app)
+    response = client.get("/")
+    assert "content-length" not in response.headers
+
+
+def test_json_response_no_content_length(test_client_factory):
+    app = JSONResponse("null", status_code=204)
+    client = test_client_factory(app)
+    response = client.get("/")
+    assert "content-length" not in response.headers
```

Comment:
I'd simplify what we test here. Instead of this, I'd suggest we just add the following case, after the existing `test_empty_response` case...

```python
def test_empty_204_response(test_client_factory):
    app = Response(status_code=204)
    client: TestClient = test_client_factory(app)
    response = client.get("/")
    assert "content-length" not in response.headers
```

This helps make the difference between the 200 and the 204 case clearer. Also it's a valid, sensible, thing to do, which these two cases aren't. (they *are* including a body, but setting status codes that shouldn't have one.)